src/api/app/datatables/package_datatable.rb

```diff
@@ -0,0 +1,38 @@
+# NOTE: Folowing: https://github.com/jbox-web/ajax-datatables-rails#using-view-helpers
+class PackageDatatable < AjaxDatatablesRails::ActiveRecord
+  extend Forwardable
+
+  def_delegator :@view, :link_to
+  def_delegator :@view, :package_show_path
+  def_delegator :@view, :time_ago_in_words
+
+  def initialize(params, opts = {})
+    @project = opts[:project]
+    @view = opts[:view_context]
+    super
+  end
+
+  def view_columns
+    # Declare strings in this format: ModelName.column_name
+    # or in aliased_join_table.column_name format
+    @view_columns ||= {
+      name: { source: 'Package.name', cond: :like },
+      changed: { source: 'Package.updated_at', cond: :like }
+    }
+  end
+
+  # rubocop:disable Naming/AccessorMethodName
+  def get_raw_records
+    @project.packages.order_by_name
+  end
+  # rubocop:enable Naming/AccessorMethodName
+
+  def data
+    records.map do |record|
+      {
+        name: link_to(record.name, package_show_path(package: record, project: record.project)),
```

Comment:
using `@project` instead `record.project` should save us many sql queries :thinking:

Reply:
Not really. This is taken care of by rails caching. The only query that happens here is loading the packages.

Reply:
Even if Rails catch the SQL query, still Rails need to check it is cached, while you have already this info stored in a variable available here.

Reply:
I am very reluctant to do micro-optimizations, but since I know that this isn't leading us anywhere I change it...